public/app/plugins/datasource/grafana-azure-monitor-datasource/insights_analytics/insights_analytics_datasource.ts

```diff
@@ -0,0 +1,32 @@
+import { ScopedVars } from '@grafana/data';
+import { DataSourceInstanceSettings } from '@grafana/data';
+import { getTemplateSrv } from '@grafana/runtime';
+
+import { AzureDataSourceJsonData, AzureMonitorQuery, AzureQueryType } from '../types';
+import AppInsightsDatasource from '../app_insights/app_insights_datasource';
+
+export default class InsightsAnalyticsDatasource extends AppInsightsDatasource {
+  constructor(instanceSettings: DataSourceInstanceSettings<AzureDataSourceJsonData>) {
+    super(instanceSettings);
+  }
+
+  applyTemplateVariables(target: AzureMonitorQuery, scopedVars: ScopedVars): Record<string, any> {
+    const item = target.insightsAnalytics;
+
+    // Old name migrations
+    const old: any = item;
+    if (old.rawQueryString && !item.query) {
+      item.query = old.rawQueryString;
+    }
+
+    return {
+      refId: target.refId,
+      queryType: AzureQueryType.InsightsAnalytics,
+      insightsAnalytics: {
+        type: 'timeSeriesQuery',
```

Comment:
nit: could we use a constant here?
```suggestion
        type: TIME_SERIES_QUERY,
```

Reply:
removed